### ISSUE TITLE
user list - pagination doesn't work correctly with SEF-URL

### ISSUE THREAD

**Issue by Jimmi08** [e107inc/e107#4253]
## Bug Description
with /user/ pagination doesn't work 
with /user.php it works

 with user.php:
/user.php?30.30.DESC

with /user/:
user/?30.30.DESC - it still displays first page.

**Comment by CaMer0n** [comment]
@Jimmi08 Does pagination with `/user/?5.5.DESC` work correctly for you?

**Comment by Jimmi08** [comment]
@CaMer0n 
I can't see any difference. 
 
/user/  - I set 5 records for page, click show, 5 users are displayed
click on page n.2
link changed to /user/?5.5.DESC
but still the first page and 20 users are displayed.

But not very e107 latest version

**Comment by CaMer0n** [comment]
@Jimmi08 Okay thank you. Unfortunately, I'm unable to reproduce the problem you're having.

**Comment by Jimmi08** [comment]
$_GET is correct, but e_QUERY is string(0) "". Weird. Only with /user/, user.php is OK.

**Comment by Jimmi08** [comment]
@CaMer0n What is your URL configuration for the user page? 
- if I let it as user.php?id=1, then /user/ - pagination works, but user profile link is legacy
So what is your to test this?

**Comment by CaMer0n** [comment]
@Jimmi08 Yes, when I switched to Friendly URL (ie. 'user/UserDisplayName') profile the pagination fails. 

I'm not sure of exactly how to fix it since it's not my code and the intended method to handle pagination seems incomplete. 

I believe the issue occurs in this file: `e107_core\url\user\rewrite_url.php` 
(btw. URL profiles need to be re-saved to clear caching of it) 

It appears that `/user/list/` was the intended URL for listing the users, but the query for pagination is unclear.

**Comment by Jimmi08** [comment]
@CaMer0n 
what about solution>
 /user/xy -  you are navigated between users (one user per page)
and 
/users/xy - paginated list of users, you are navigating between pages

I understand that now it is messed up because it is not easy to detect what page you are on with the same URL.
Thanks

**Comment by Jimmi08** [comment]
This is still an issue.   You can't use pretty URLs for users then.

**Comment by Jimmi08** [comment]
I think that the first problem is using e_SELF for form records and outdated pagination , not rewriting itself.

**Comment by 3l3ktr0n** [comment]
has anyone found a fix for this by now?

**Comment by Jimmi08** [comment]
@3l3ktr0n thank you for reminding this issue, I almost forget it,

This is not fix, problem is deeper, it is just workaround,

open user.php
find lines (cca 144)
```
if (isset($_POST['records']))
{
	$records = intval($_POST['records']);
	$order = ($_POST['order'] == 'ASC' ? 'ASC' : 'DESC');
	$from = 0;
}
```
and change them to:  (different from and POST to REQUEST)
```
if (isset($_REQUEST['records']))
{
	$records = intval($_REQUEST['records']);
	$order = ($_REQUEST['order'] == 'ASC' ? 'ASC' : 'DESC');
	$from = intval($_REQUEST['from']);
}
```

Find lines
```
if (vartrue($records) > 50)
{
	$records = 50;
}
```
and add - this is just check to avoid dividing by zero (it is always risk with POST parameters and this is not checked in pagination itself where is correct place for it)

```
if (vartrue($records) < 5)
{
	$records = 5;
}
```

find line:
`$parms = $users_total.",".$records.",".$from.",".e_SELF.'?[FROM].'.$records.".".$order;`

and change it:
`$parms = $users_total.",".$records.",".$from.",".e_SELF.'?from=[FROM]&records='.$records."&order=".$order;`

Their is already form for setting those stuff (item per page and order) but it works with $_GET. So I added them as $_POST too 
:) 

The main issue still left. again THIS IS NOT CORRECT SOLUTION. 
 e_QUERY is not working//filled and $_GET are changed to X_X_X without any naming...   
I just changed it on my site, so you can test this if you want.

**Comment by Jimmi08** [comment]
It is still an issue. Tested on e107.nl.
